client/js/genre.js

```diff
@@ -1,24 +1,23 @@
 /**
- * fetches youtube api calls from YoutubeServlet.java
+ * @file fetches youtube api calls from YoutubeServlet.java
  * and displays on youtube-genre.html
  */
 
 
 /**
- * fetches genre count hashmap from /api/youtube and updates html
+ * fetches and returns genre analysis object from /api/youtube
+ *
+ * @returns {Promise<object>} obj containing Youtube genre data and stats
  */
-async function displayMusicGenre() {
-  const genreBlock = document.getElementById('genres');
-
-  // keep track of num_videos in URL w/o reload
-  history.pushState('', '', `youtube-genre.html`);
-
-  const response = await fetch(`/api/youtube`);
+async function fetchMusicGenre() {
+  const response = await fetch('/api/youtube');
   if (response.status == 401) {
     // no oauth login so redirect to new page
     window.open('/api/oauth/login/youtube');
   }
 
   const genreCount = await response.text();
-  genreBlock.innerHTML = genreCount;
+  return JSON.parse(genreCount);
 }
+
+export const GENRE_ANALYSIS_PROMISE = fetchMusicGenre();
```

Comment:
See earlier PR about not calling a function with side effects (such as making a web request) on import